tests/test_update_ssh_keys.rs

```diff
@@ -0,0 +1,705 @@
+            File::options().create(true).append(true).write(true).open(authkeys.clone()).expect(format!("unable to create a file {}", authkeys.to_str().unwrap_or_default()).as_str());
+        for (_, text) in &*TEST_KEYS {
+            _ = authkeysfile.write_all(format!("{}\n", text).as_bytes());
+        }
+
+        // equivalent of running "update-ssh-keys" without args
+        let aks = open_authorized_keys(ssh_dir);
+
+        aks.write().expect("failed to update authorized keys directory");
+        aks.sync().expect("failed to update authorized keys");
+
+        let authkeys: PathBuf = PathBuf::from(format!("{}/authorized_keys.d/old_authorized_keys", ssh_dir));
+        assert!(authkeys.exists());
+        assert_has_keys(ssh_dir, ["valid1", "valid2", "valid3"].to_vec(), true);
+    }
+
+    #[test]
+    fn test_add_one_file() {
+        let ctx = setup_tests();
+
+        let ssh_dir = ctx.ssh_dir.to_str().unwrap();
+        let pubkeyvalid1 = "valid1";
+        let testnameone = "one";
+        let assert_keys = ["valid1"].to_vec();
+
+        // "update-ssh-keys --add one valid1.pub"
+        match add_one_ssh_key(
+            ssh_dir,      // ssh_dir
+            pubkeyvalid1, // pubkeyname
+            testnameone,  // testname
+            false,        // is_force
+            true,         // is_replace
+        ) {
+            Ok(keys) => add_key_check_results(ssh_dir, keys, pubkeyvalid1, testnameone, assert_keys),
+            Err(err) => panic!("update_ssh_keys --add failed {}", err),
+        }
+    }
+
+    #[test]
+    fn test_replace_one() {
+        let ctx = setup_tests();
+
+        let ssh_dir = ctx.ssh_dir.to_str().unwrap();
+        let pubkeyvalid1 = "valid1";
+        let pubkeyvalid2 = "valid2";
+        let testnameone = "one";
+        let assert_keys = ["valid1"].to_vec();
+
+        // "update-ssh-keys --add one valid1.pub"
+        match add_one_ssh_key(
+            ssh_dir,      // ssh_dir
+            pubkeyvalid1, // pubkeyname
+            testnameone,  // testname
+            false,        // is_force
+            true,         // is_replace
+        ) {
+            Ok(keys) => add_key_check_results(ssh_dir, keys, pubkeyvalid1, testnameone, assert_keys),
+            Err(err) => panic!("update_ssh_keys --add failed {}", err),
+        }
+
+        let assert_keys = ["valid2"].to_vec();
+
+        // "update-ssh-keys --add one valid2.pub"
+        match add_one_ssh_key(
+            ssh_dir,      // ssh_dir
+            pubkeyvalid2, // pubkeyname
+            testnameone,  // testname
+            false,        // is_force
+            true,         // is_replace
+        ) {
+            Ok(keys) => add_key_check_results(ssh_dir, keys, pubkeyvalid2, testnameone, assert_keys),
+            Err(err) => panic!("update_ssh_keys --add failed {}", err),
+        }
+    }
+
+    #[test]
+    fn test_no_replace() {
+        let ctx = setup_tests();
+
+        let ssh_dir = ctx.ssh_dir.to_str().unwrap();
```

Comment:
Code below can be replaced by a call to `run_test_add_one_file(&ctx)`

Reply:
Called `run_test_add_one_file(&ctx.ssh_dir)`.